app/controllers/application_controller/wait_for_task.rb

```diff
@@ -11,65 +11,64 @@ def wait_for_task
     @edit = session[:edit]  # If in edit, need to preserve @edit object
     raise Forbidden, _('Invalid input for "wait_for_task".') unless params[:task_id]
 
-    @edit = session[:edit]  # If in edit, need to preserve @edit object
-    session[:async] ||= {}
-    session[:async][:interval] ||= 1000 # Default interval to 1 second
-    session[:async][:params] ||= {}
+    async_interval = params[:async_interval] || 1000 # Default interval to 1 second
+    async_params = params[:async_params] || session[:async][:params]
 
     if MiqTask.find(params[:task_id].to_i).state != "Finished" # Task not done --> retry
-      browser_refresh_task(params[:task_id])
+      browser_refresh_task(params[:task_id], async_params, async_interval)
     else # Task done
-      session[:async][:params].each { |k, v| @_params[k] = v } # Merge in the original params and
-      send(session.fetch_path(:async, :params, :action)) # call the orig. method
+      async_params.each { |k, v| @_params[k] = v } # Merge in the original params and
+      send(async_params[:action]) # call the orig. method
     end
   end
 
-  def browser_refresh_task(task_id, should_flash = false)
-    session[:async][:interval] += 250 if session[:async][:interval] < 5000 # Slowly move up to 5 second retries
+  def browser_refresh_task(task_id, async_params, async_interval, should_flash: false)
+    async_interval += 250 if async_interval < 5000 # Slowly move up to 5 second retries
     render :update do |page|
       page << javascript_prologue
-      ajax_call = remote_function(:url => {:action => 'wait_for_task', :task_id => task_id})
-      page << "setTimeout(\"#{ajax_call}\", #{session[:async][:interval]});"
+      ajax_call = remote_function(:url => {:action => 'wait_for_task', :task_id => task_id, :async_params => async_params, :async_interval => :async_interval})
```

Comment:
Oh because of this ... hmmmm

Reply:
async_parameters have been removed. now it only contains async_interval